C_CPP/Competitive Programming/Find_Day_of_a_given_Date.md

```diff
@@ -0,0 +1,115 @@
+
+Then find the day as per following table:    
+| SUNDAY |	MONDAY | TUESDAY |	WEDNESDAY |	THURSDAY |	FRIDAY |	SATURDAY |      
+| ------ | ------- | ------- | ---------- | -------- | ------- | --------- |   
+| 0	     | 1       | 2	     | 3	        | 4        | 5       |	6        |   
+
+### IMPLEMENTATION IN CPP:  
+Now we implement the above approach in cpp using OOPS concept:  
+```cpp
+#include <iostream>
+#include <conio.h>
+using namespace std;
+class Day
+{
+      public:
+             int k,m,D,C,F,R;
+             
+                void input_elements()
+                {
+                    cout<<"enter the date number"<<endl;
+                    cin>>k;
+                    cout<<"enter the month number"<<endl;
+                    cin>>m;
+                    cout<<"enter the first two digits of the year"<<endl;
+                    cin>>C;
+                    cout<<"enter the last two digits of the year"<<endl;
+                    cin>>D;
+                }
+             void Cal()
+             {
+                m=m-2;
+                if(m <=0)
+                {
+                    m=m+12;
+                    D=D-1;
+                }
+                     
+                F= k+ (int)13*m-1/12+D + (int)D/4 + (int)C/4-2*C;
+                if(F >= 0)
+                {
+                    R=F%7;
+                }
+                else if(F<0)
+                {
+                    R=7-(F%(-7));
+                }
+                    
+                          
+                switch(R)
+                {
+                    case 0:
+                        cout<<"Sunday"<<endl;
+                        break;
+                    case 4:
+                        cout<<"Monday"<<endl;
+                        break;
+                    case 5:
+                        cout<<"Tuesday"<<endl;
+                        break;
+                    case 6:
+                        cout<<"Wednesday"<<endl;
+                        break;
+                    case 1:
+                        cout<<"Thursday"<<endl;
+                        break;
+                    case 2:
+                        cout<<"Friday"<<endl;
+                        break;
+                    case 3:
+                        cout<<"Saturday"<<endl;
+                        break;
+                    default:
+                        cout<<"try again"<<endl;
+                        break;
+                }
+             }
+             
+};
+
+      
+int main()
+{
+    Day t;
+    t.input_elements();
+    t.Cal();
+}
+ ```  
+### OUTPUT:  
+Here we calculate day for 9 November 2021 .  
+k= 9  
+m= 11  
+C= 20  
+D= 21  
+Desired changes have been made in code for finding the correct output.  
+
+ ![image](https://user-images.githubusercontent.com/77229404/143466278-57fe1b53-f6c9-45b1-bad1-652603d40e6b.png)
+
+### REFERENCES:
+-https://beginnersbook.com/2013/04/calculating-day-given-date/
```

Comment:
[Beginnersbook](https://beginnersbook.com/2013/04/calculating-day-given-date/)

put the link like this